apps/saleor-app-checkout/__tests__/backend/payments/providers/adyen/utils.test.ts

```diff
@@ -2,53 +2,379 @@ import {
 
-      const key1 = createEventUniqueKey(event);
-      const key2 = createEventUniqueKey(event);
+describe("createEventUniqueKey", () => {
+  it("creates the same key for single event", () => {
+    const event: TransactionEventFragment = {
+      reference: "ref123",
+      name: "event",
+    };
 
-      expect(key1).toEqual(key2);
+    const key1 = createEventUniqueKey(event);
+    const key2 = createEventUniqueKey(event);
+
+    expect(key1).toEqual(key2);
+  });
+
+  it("creates different keys for different events", () => {
+    const event1: TransactionEventFragment = {
+      reference: "ref123",
+      name: "event1",
+    };
+    const event2: TransactionEventFragment = {
+      reference: "ref123",
+      name: "event2",
+    };
+
+    const key1 = createEventUniqueKey(event1);
+    const key2 = createEventUniqueKey(event2);
+
+    expect(key1).not.toEqual(key2);
+  });
+});
+
+describe("getSaleorAmountFromAdyen", () => {
+  it("parses integer into float value", () => {
+    expect(getSaleorAmountFromAdyen(1922)).toBe(19.22);
+    expect(getSaleorAmountFromAdyen(1000)).toBe(10);
+    expect(getSaleorAmountFromAdyen(837)).toBe(8.37);
+  });
+});
+
+describe("getAdyenAmountFromSaleor", () => {
+  it("parses float into integer value", () => {
+    expect(getAdyenAmountFromSaleor(19.22)).toBe(1922);
+    expect(getAdyenAmountFromSaleor(10)).toBe(1000);
+    expect(getAdyenAmountFromSaleor(8.37)).toBe(837);
+  });
+});
+
+describe("getTransactionAmountFromAdyen", () => {
+  const DEFAULT_CURRENCY = "EUR";
+  const prepareAdyenNotification = (
+    amount: number,
+    eventCode: EventCodeEnum
+  ): NotificationRequestItem => ({
+    amount: {
+      currency: DEFAULT_CURRENCY,
+      value: amount,
+    },
+    eventCode,
+    pspReference: "OPERATION ID",
+    success: SuccessEnum.True,
+    additionalData: {},
+    eventDate: new Date().toString(),
+    merchantReference: "ORDER",
+    merchantAccountCode: "Saleor",
+    reason: "reason",
+    operations: [],
+    paymentMethod: "visa",
+    originalReference: "PAYMENT ID",
+  });
+
+  it("returns nothing when notification is failed", () => {
+    const notification = {
+      ...prepareAdyenNotification(12_00, EventCodeEnum.Authorisation),
+      success: SuccessEnum.False,
+    };
+
+    const result = getTransactionAmountFromAdyen(notification, null);
+
+    expect(result).toStrictEqual({});
+  });
+
+  it("returns nothing if payment is pending", () => {
+    const notification = prepareAdyenNotification(12_00, EventCodeEnum.Pending);
+    const result = getTransactionAmountFromAdyen(notification, null);
+
+    expect(result).toStrictEqual({});
+  });
+
+  it("throws an error if amount is missing", () => {
+    const notification = {
+      ...prepareAdyenNotification(12_00, EventCodeEnum.Authorisation),
+      amount: undefined,
+    };
+
+    expect(() => {
+      return getTransactionAmountFromAdyen(
+        // @ts-expect-error
+        notification,
```

Comment:
Is it possible to assert the type here? As in:
```suggestion
        notification as Types.notification.NotificationRequestItem,
```

This way if we ever modify the implementation of `getTransactionAmountFromAdyen` there's higher chance we'll spot the error.

Reply:
no, we can't assert *in this place* - because we check for runtime errors TypeScript will catch. I've added however an assertion on the fields we don't check for errors (so everything, except `amount`)

Reply:
I don't think I understand what you mean by `runtime errors TypeScript will catch`. Can you elaborate?

Reply:
Ah, sorry - the previous answer doesn't really make sense:
- TypeScript will check statically errors in our code, in this line of test it will raise an issue because we're passing a notification object **without amount**.
- We do that on purpose, we want to check if the function handles a case when amount is missing **in runtime**

Reply:
Gotcha! So the error is actually expected.